api/diff.go

```diff
@@ -78,37 +84,49 @@ func handleAPIDiffGet(w http.ResponseWriter, r *http.Request) {
 	diffFilter, paths, err := shared.ParseDiffFilterParams(q)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusBadRequest)
+
 		return
 	}
 
 	runs, err := loadDiffRuns(store, q)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusBadRequest)
+
 		return
 	}
 	if runs == nil {
 		http.NotFound(w, r)
+
 		return
 	}
 
 	if len(runs) != 2 {
 		http.Error(w, fmt.Sprintf("Diffing requires exactly 2 runs, but found %v", len(runs)), http.StatusBadRequest)
+
 		return
 	}
 
 	diffAPI := shared.NewDiffAPI(ctx)
 	diff, err := diffAPI.GetRunsDiff(runs[0], runs[1], diffFilter, paths)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
+
 		return
 	}
 
 	var bytes []byte
 	if bytes, err = json.Marshal(diff); err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
+
 		return
 	}
-	w.Write(bytes)
+	_, err = w.Write(bytes)
+	// nolint:godox // TODO: Golangci-lint found that we previously ignored the error.
+	// We should investigate if we should return a HTTP error or not. In the meantime, we log the error.
+	if err != nil {
+		logger := shared.GetLogger(r.Context())
+		logger.Warningf("Failed to write data: %w", err.Error())
```

Comment:
```suggestion
		logger.Warningf("Failed to write data in diff API: %w", err.Error())
```

This will allow us to easily know where this was printed.